oscilloscope/oscilloscope_server.py

```diff
@@ -0,0 +1,227 @@
+"""
+### BEGIN NODE INFO
+[info]
+name = Oscilloscope
+version = 0.1
+description = Talks to oscilloscopes
+
+[startup]
+cmdline = %PYTHON% %FILE%
+timeout = 20
+
+[shutdown]
+message = 987654321
+timeout = 20
+### END NODE INFO
+"""
+
+from __future__ import absolute_import
+from labrad.server import setting
+from labrad.gpib import GPIBManagedServer
+from twisted.internet.defer import inlineCallbacks, returnValue
+
+
+import oscilloscope.tektronix as tektronix
+import oscilloscope.agilent as agilent
+
+
+class OscilloscopeServer(GPIBManagedServer):
+    """Manges communication with oscilloscopes. ALL the oscilloscopes."""
+
+    name = 'Oscilloscope Server'
+
+    deviceWrappers = {
+            'AGILENT TECHNOLOGIES DSO-X 4104A': agilent.AgilentDSOX4104AWrapper}
+
+    # SYSTEM
+
+    @setting(11, returns='')
+    def reset(self, c):
+        """Reset the oscilloscope to factory settings."""
+        dev = self.selectedDevice(c)
+        yield dev.reset()
+
+    @setting(12, returns='')
+    def clear_buffers(self, c):
+        """Clear device status buffers."""
+        dev = self.selectedDevice(c)
+        yield dev.clear_buffers()
+
+    # CHANNEL
+
+    @setting(50, channel='i', state='s', returns='s')
+    def channel_on_off(self, c, channel, state=None):
+        dev = self.selectedDevice(c)
+        result = yield dev.channel_on_off(channel, state)
+        returnValue(result)
+
+    @setting(51, channel='i', coup='s', returns='s')
+    def coupling(self, c, channel, coup=None):
+        """Set or query channel coupling.
+
+        Args:
+            channel (int): Which channel to set coupling.
+            coup (str): Coupling (usually 'AC' or 'DC'). If None (the default)
+                just query the coupling without setting it.
+
+        Returns:
+            string indicating the channel's coupling.
+        """
+        dev = self.selectedDevice(c)
+        result = yield dev.coupling(channel, coup)
+        returnValue(result)
+
+    @setting(52, channel='i', invert='i', returns='i')
+    def invert(self, c, channel, invert=None):
+        """Get or set channel inversion.
+
+        Args:
+            channel (int):
+            invert (int): 1 -> invert channel, 0 -> do not invert channel. If
+                None (the default), then we only query the inversion.
+
+        Returns:
+            (int): 0: not inverted, 1: inverted.
+        """
+        dev = self.selectedDevice(c)
+        result = yield dev.invert(channel, invert)
+        returnValue(result)
+
+    @setting(53, channel='i', term='i', returns='i')
+    def termination(self, c, channel, term=None):
+        """Set channel termination
+
+        Args:
+            channel (int): Which channel to set termination.
+            term (int): Termination in Ohms. Either 50 or 1,000,000 on every
```

Comment:
Definitely not using `v[Ohm]` as floating points don't make sense here. The string version could work, and perhaps makes more sense.